code/assurance/src/main/java/com/adobe/marketing/mobile/assurance/EventStitcher.kt

```diff
@@ -0,0 +1,131 @@
+        this.queue = queue
+        this.notifier = notifier
+    }
+
+    /**
+     * Receives AssuranceEvents and stitches them if they are chunked and notifies the notifier with the stitched event.
+     * If the event is not chunked, it is directly notified via the notifier.
+     *
+     * @param event the AssuranceEvent
+     */
+    @JvmName("onEvent")
+    internal fun onEvent(event: AssuranceEvent) {
+        if (!isChunked(event)) {
+            notifier.call(event)
+            return
+        }
+
+        val chunkId: String = event.metadata[AssuranceConstants.AssuranceEventKeys.CHUNK_ID] as String? ?: return
+        val totalChunks: Int = event.metadata[AssuranceConstants.AssuranceEventKeys.CHUNK_TOTAL] as Int? ?: return
+
+        // check if there are prior chunks associated with this chunkId in the queue, if not create a new queue
+        val chunkedEventsForId: MutableList<AssuranceEvent> = queue[chunkId] ?: mutableListOf()
+        chunkedEventsForId.add(event)
+
+        // if all chunks are received, stitch the events and notify the caller
+        if (chunkedEventsForId.size == totalChunks) {
+            val result: Response<AssuranceEvent, Exception> = stitch(chunkedEventsForId)
+            when (result) {
+                is Response.Success -> notifier.call(result.data)
+                is Response.Failure -> Log.error(Assurance.LOG_TAG, LOG_TAG, "Failed to stitch events for chunkId: $chunkId due to: ${result.error.message}")
+            }
+            // remove the event from the queue after stitching
+            queue.remove(chunkId)
+        } else {
+            // else add the event to the queue for stitching once all pieces are received
+            queue[chunkId] = chunkedEventsForId
+        }
+    }
+
+    /**
+     * Checks whether the event is chunked.
+     * @param event the event to check
+     * @return true if the event is chunked, false otherwise
+     */
+    @JvmName("isChunked")
+    internal fun isChunked(event: AssuranceEvent): Boolean {
+        event.metadata?.let {
+            val chunkId = it[AssuranceConstants.AssuranceEventKeys.CHUNK_ID] as String?
+            return chunkId != null
+        } ?: return false
+    }
+
+    /**
+     * Stitches the chunked events and returns the stitched event as part of the response.
+     * @param chunkedEvents the list of chunked events to stitch
+     * @return a stitched event as part of [Response.Success] or an exeption as part of [Response.Failure]
+     */
+    @VisibleForTesting
+    internal fun stitch(chunkedEvents: MutableList<AssuranceEvent>): Response<AssuranceEvent, Exception> {
+        if (chunkedEvents.isEmpty()) return Response.Failure(Exception("No events to stitch"))
+
+        Log.trace(Assurance.LOG_TAG, LOG_TAG, "Stitching ${chunkedEvents.size} events")
+
+        // sort the events by sequence number
+        chunkedEvents.sortBy { it.metadata[AssuranceConstants.AssuranceEventKeys.CHUNK_SEQUENCE_NUMBER] as Int }
```

Comment:
This could throw an exception. Can you validate a valid `CHUNK_SEQUENCE_NUMBER` exists before queueing in [onEvent](https://github.com/adobe/aepsdk-assurance-android/pull/62/files#diff-fc526232d727fc11ef6285713259ba7170970e3684b5b87273e8d9b76da3a8b3R61) function